src/metabase/util.clj

```diff
@@ -423,6 +423,21 @@
   (^String [s max-length]
    (str/join (take max-length (slugify s)))))
 
+(defn slugify-filename
```

Comment:
Ok, so this is exactly the same as the existing `u/slugify` except for the behavior for non-ASCII characters, right? 

Two thoughts:

1. There is nothing special about this version of the function that makes it filename specific other than the place you're using it. I don't think `-filename` is relevant here. `slugify-without-encoding-non-ascii` or something would actually describe **what** the function does, rather than its current use case. Whether you're using it for a filename or not is not relevant and for all we know maybe we **do** want to encode non-ASCII characters in some situation for some filename somewhere. Or (more likely) maybe we want to slugify something using the same rules but without using the result as a file name... why box ourselves in?
2. Instead of creating an entirely new function that does one thing slightly different, why not just refactor the existing function? One function with two options is better than two functions with one option or four functions with no options for that matter. Right now `u/slugify` has a one-arity that takes a string and a two-arity that takes a `string` and `max-length`. I'd suggest changing the two-arity to take an options map, where `:max-length` is one of the options, and then add something like an `:encode-non-ascii` option that will let you choose between the old behavior or the new. I grepped the codebase and you'll only have to update one line to use the new options map

Reply:
Done. It's now an option with `unicode? true` to preserve. So the default is still to URL-encode, as with classic `u/slugify`.